### DISCUSSION TITLE
Is it possible to have MSW Data automatically generate a related object?

### DISCUSSION THREAD

**Discussion by philcunliffe** [mswjs/data#284] [Q&A]
Let's take the user / post author example from the documentation. Is it possible to have MSW Data automatically generate a user to assign to the author variable if one isn't passed in?

**Comment by kettanaito** [comment]
Hi, @philcunliffe. At its current state, Data doesn't automatically generate any end of a declared relationship. You have to associate models manually. The relationship itself acts as a type and runtime guard of the values you provide to the relational property (so you don't associate `pet` to `user.posts` when you expect it to be `manyOf('post')`).

```js
const db = factory({
  user: { 
    id: primaryKey(String),
    posts: manyOf('post')
  },
  post: {
    id: primaryKey(String),
    title: String
  }
})

db.author.create({
  posts: [
    db.post.create({ title: 'First post' }),
    db.post.create({ title: 'Second post' })
  ]
})
```

While this works for simple relationships, I acknowledge that it becomes problematic to manually associate many-to-one relationships, for example, where the nature of a relationship is circular (`user.posts: manyOf('post')` while `post.author: oneOf('user')`). 

I'm working on the next version of this library that already addresses this issue and introduces something called _inversed relationships_. If you have any particular use cases to share about how and when you want to use relationships that automatically generate values, please do so below. Thanks.
